### ISSUE TITLE
serviceaccount.create = false doesn't work

### ISSUE THREAD

**Issue by acuteaura** [netdata/helmchart#206]
the SA name is hardcoded into the deployment regardless, so all you get it a stuck replicaset

**Comment by ilyam8** [comment]
Hi @acuteaura 

Let's see [our docs](https://github.com/netdata/helmchart#configuration)

> serviceAccount.create
> if true, create a service account

Our manifests

## serviceaccount.yaml

We create `ServiceAccount` only if `serviceAccount.create` is `true` - so far our docs are correct.

https://github.com/netdata/helmchart/blob/7f32dada6c881122ecbab79ec19df398dea3cfa8/charts/netdata/templates/serviceaccount.yaml#L1-L11

## clusterrolebinding.yaml

Regardless `serviceAccount.create` we bind `ClusterRole` to `serviceAccount.name`.

https://github.com/netdata/helmchart/blob/7f32dada6c881122ecbab79ec19df398dea3cfa8/charts/netdata/templates/clusterrolebinding.yaml#L15-L17

## deployment.yaml (parent)

Regardless `serviceAccount.create` we set `serviceAccountName` to `serviceAccount.name`.

https://github.com/netdata/helmchart/blob/7f32dada6c881122ecbab79ec19df398dea3cfa8/charts/netdata/templates/deployment.yaml#L36

## daemonset.yaml (child)

Regardless `serviceAccount.create` we set `serviceAccountName` to `serviceAccount.name`.

https://github.com/netdata/helmchart/blob/7f32dada6c881122ecbab79ec19df398dea3cfa8/charts/netdata/templates/daemonset.yaml#L39

---

It looks like `serviceAccount.create: false` doesn't really work, you are right.

@acuteaura what is your goal? Get Netdata running and working with default service account?

**Comment by acuteaura** [comment]
We use only the parent netdata as a streaming endpoint with all collectors disabled for machines doing CI outside a cluster and then scrape the data with prometheus, so there is no need for our instance to ever call the kubernetes API.

**Comment by ilyam8** [comment]
So your setup is parent installed inside k8s and children directly on hosts and they stream to the parent?

If so, perhaps this will do:
 - rbac.create: false
 - serviceAccount.create: false
 - serviceAccount.name: default

**Comment by acuteaura** [comment]
That'd work, I guess I just expected it to work differently.

Unfortunately our use case went far far outside the scope of this helm chart, so I had to fork and hardcode some things. I'm hoping I can generalize some of those changes eventually (multiple ports, mostly) and upstream it eventually.

Thanks for your time.

**Comment by ilyam8** [comment]
> our use case went far far outside the scope of this helm chart, so I had to fork and hardcode some things.

Have you considered creating an issue(s)? 

If you don't feel like doing it now - would be nice to know what you had to hardcode, can you tell?. I am really curious!

**Comment by acuteaura** [comment]
> > our use case went far far outside the scope of this helm chart, so I had to fork and hardcode some things.
> 
> Have you considered creating an issue(s)?
> 
> If you don't feel like doing it now - would be nice to know what you had to hardcode, can you tell? I am really curious!

We need to allow streaming from the internet, while restricting access to the other parts that are usually on the same listener. We figured out to late streaming was not actually encapsulated by HTTP, so we couldn't use our usual authentication mechanisms (IAP Proxy). ACLs weren't a great solution either because the service CIDR range is not always the same and Google LBs - if they don't use the PROXY protocol - also originate their traffic in 10.0.0.0/8, so we added a second port to the deployment for things like readiness/liveness, prometheus scraping or debug access to the dashboard that is not exposed.
